hmt_subgraph/src/mapping/hm-token.ts

```diff
@@ -149,3 +158,42 @@ export function handleBulkApproval(event: BulkApproval): void {
 
   entity.save();
 }
+
+function updateHolders(
+  holderAddress: Address,
+  value: BigInt,
+  increase: boolean
+): number {
+  if (
+    holderAddress.toHexString() == "0x0000000000000000000000000000000000000000"
+  )
+    return 0;
+  let count = 0;
+  let id = holderAddress.toHex();
+  let holder = Holder.load(id);
+  const isNew = holder == null ? true : false;
+
+  if (holder == null) {
+    holder = new Holder(id);
+    holder.address = holderAddress;
+    holder.balance = BigInt.fromI32(0);
+  }
+  const balanceBeforeTransfer = holder.balance;
+  holder.balance = increase
+    ? holder.balance.plus(value)
+    : holder.balance.minus(value);
+
+  if (
+    (isNew && !holder.balance.isZero()) ||
+    (!isNew && balanceBeforeTransfer.isZero())
+  ) {
+    count = 1;
+  } else if (holder.balance.isZero()) {
+    count = -1;
+    store.remove("Holder", id);
+  }
```

Comment:
shouldn't be possible to have a holder which is not new and his old balance was 0.
I think this would be a simpler solution:
```suggestion
    if (isNew && !holder.balance.isZero()) {
    count = 1;
  } else if (!isNew && holder.balance.isZero()) {
    count = -1;
    store.remove("Holder", id);
  }
```

Reply:
I updated the code and removed `store.remove("Holder", id);` so we can keep the history of hodler even if their balance is 0

Reply:
The token contract allows to transfer amount 0. So in this case if the destination address balance was 0 before the value returned would be -1.

You can verify the error using this test in hmt.test.ts:
```describe("HMToken entity", () => {
  test("Should properly calculate holders in statistics", () => {
    let transfer1 = createTransferEvent(
      "0x0000000000000000000000000000000000000000",
      "0x92a2eEF7Ff696BCef98957a0189872680600a959",
      10,
      BigInt.fromI32(10)
    );

    let transfer2 = createTransferEvent(
      "0x92a2eEF7Ff696BCef98957a0189872680600a959",
      "0xD979105297fB0eee83F7433fC09279cb5B94fFC6",
      0,
      BigInt.fromI32(10)
    );

    handleTransfer(transfer1);
    handleTransfer(transfer2);

    assert.fieldEquals(
      "HMTokenStatistics",
      HMT_STATISTICS_ENTITY_ID,
      "holders",
      "1"
    );

    let transfer3 = createTransferEvent(
      "0x0000000000000000000000000000000000000000",
      "0xD979105297fB0eee83F7433fC09279cb5B94fFC6",
      10,
      BigInt.fromI32(10)
    );
    
    handleTransfer(transfer3);

    assert.fieldEquals(
      "HMTokenStatistics",
      HMT_STATISTICS_ENTITY_ID,
      "holders",
      "2"
    );

    let transfer4 = createTransferEvent(
      "0x92a2eEF7Ff696BCef98957a0189872680600a959",
      "0xD979105297fB0eee83F7433fC09279cb5B94fFC6",
      10,
      BigInt.fromI32(10)
    );

    handleTransfer(transfer4);
   
    assert.fieldEquals(
      "HMTokenStatistics",
      HMT_STATISTICS_ENTITY_ID,
      "holders",
      "1"
    );

    clearStore();
  });
});
```

_*This test should be added when fixed._

Solution:
```
if (balanceBeforeTransfer.isZero() && !holder.balance.isZero()) {
    count = 1;
  } else if (!balanceBeforeTransfer.isZero() && holder.balance.isZero()) {
    count = -1;
  }
```
And remove isNew variable (unused)

Reply:
Indeed I had not taken into account that the contract allows to transfer amount 0, I updated the the mapper and added the test. Thanks you !